src/workspaces/common/state/useCloningWorkspaceNotifications.test.tsx

```diff
@@ -0,0 +1,143 @@
+      // Arrange
+      const clone: WorkspaceInfo = {
+        ...defaultAzureWorkspace.workspace,
+        state: 'Cloning',
+      };
+      // Act
+      notifyNewWorkspaceClone(clone);
+
+      // Assert
+      expect(asMockedFn(notify)).toHaveBeenCalledWith('info', expect.any(Object), {
+        id: expect.any(String),
+        message: expect.any(Object),
+      });
+    });
+  });
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  it.each<{
+    updatedState: WorkspaceState;
+    notificationType?: NotificationType;
+  }>([
+    { updatedState: 'CloningFailed', notificationType: 'error' },
+    { updatedState: 'Ready', notificationType: 'success' },
+  ])(
+    'removes the workspace in the store and sends a $notificationType notification when the workspace is updated to $updatedState',
+    async ({ updatedState, notificationType }) => {
+      // Arrange
+      const clone: WorkspaceInfo = { ...defaultAzureWorkspace.workspace, state: 'Cloning' };
+      cloningWorkspacesStore.set([clone]);
+      const mockDetailsFn = jest.fn().mockResolvedValue({ workspace: { state: updatedState } });
+      const mockAjax: DeepPartial<AjaxContract> = {
+        Workspaces: {
+          workspace: () =>
+            ({
+              details: mockDetailsFn,
+            } as Partial<AjaxWorkspacesContract['workspace']>),
+        },
+      };
+      asMockedFn(Ajax).mockImplementation(() => mockAjax as AjaxContract);
+
+      // Act
+      render(<CloningTestComponent />);
+      jest.advanceTimersByTime(30000);
+      await waitFor(() => expect(mockDetailsFn).toBeCalledTimes(1));
+
+      // Assert
+      expect(cloningWorkspacesStore.get()).toHaveLength(0);
+      expect(asMockedFn(clearNotification)).toHaveBeenCalledWith(
+        expect.stringContaining(`${clone.namespace}/${clone.name}`)
+      );
+      expect(asMockedFn(notify)).toHaveBeenCalledWith(notificationType, expect.any(Object), { id: expect.any(String) });
+    }
+  );
+
+  it('updates the workspace when the state is updated to to CloningContainer', async () => {
+    // Arrange
+    const clone: WorkspaceInfo = { ...defaultAzureWorkspace.workspace, state: 'Cloning' };
+    cloningWorkspacesStore.set([clone]);
+    const mockDetailsFn = jest.fn().mockResolvedValue({ workspace: { state: 'CloningContainer' } });
+    const mockAjax: DeepPartial<AjaxContract> = {
+      Workspaces: {
+        workspace: () =>
+          ({
+            details: mockDetailsFn,
+          } as Partial<AjaxWorkspacesContract['workspace']>),
+      },
+    };
+    asMockedFn(Ajax).mockImplementation(() => mockAjax as AjaxContract);
+    jest.useFakeTimers();
+
+    // Act
+    render(<CloningTestComponent />);
+    jest.advanceTimersByTime(30000);
+    await waitFor(() => expect(mockDetailsFn).toBeCalledTimes(1));
+
+    // Assert
+    expect(cloningWorkspacesStore.get()).toHaveLength(1);
+    expect(cloningWorkspacesStore.get()[0].state).toBe('CloningContainer');
```

Comment:
Is there a reason why we update the state? I can see that perhaps it would be informative to change the message at that point, but we don't actually do that.

Reply:
Keeping it at the current state makes it easier to add behavior later, simplifies the logic (so that the callbacks are always checked if the state has changed at all), and means we don't have to add some sort of exception to whether or not we update to the new state.